codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -627,6 +624,16 @@ final void close() {
         private ByteBuf headerBlock;
 
         /**
+         * The local header size maximum has been exceeded while accumulating bytes.
+         * @throws Http2Exception A connection error indicating too much data has been received.
+         */
+        private void headerSizeExceeded() throws Http2Exception {
```

Comment:
Those methods are outside the scope of `HeadersBlockBuilder` and I would have to pass an instance in as a parameter to be able to do the `close()` method.  This method is not doing any `verify`...the check has been done outside the scope and is triggered in different conditions depending on the state of the `HeadersBlockBuilder`.  I could move...but not sure if it provides any benefit. WDTY?